src/Temporalio/Workflows/Mutex.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Temporalio.Workflows
+{
+    /// <summary>
+    /// Mutex is a thin wrapper around <see cref="Semaphore"/> with a single count.
```

Comment:
Probably mention "And it is a safe alternative to System.Threading.Mutex"

I'd add that language to the semaphores too if they don't have it.

Reply:
Yes, it was already on semaphore. Added it here.